torch_xla/csrc/xla_sharding_util.h

```diff
@@ -118,6 +118,16 @@ class ShardingUtil {
       const at::Tensor& tensor, const XLATensor::ShardingSpecPtr shardings,
       const std::vector<std::string>& devices, bool padded = true);
 
+  // Retrieve output sharding of a given XLA computation.
+  static std::vector<XLATensor::ShardingSpecPtr> GetOutputSharding(
+      std::vector<xla::Shape>* output_shapes, ComputationPtr computation,
+      const torch::lazy::BackendDevice& device);
+
+  // Create sharded data placeholders, each corresponding to the individual
+  // sharding spec from the input list
+  static std::vector<torch::lazy::BackendDataPtr> CreateShardedPlaceholder(
+      const std::vector<XLATensor::ShardingSpecPtr>& sharding_specs);
+
```

Comment:
oh I forgot to remove `PrepareOutputShardingPropagation ` below since it is not being used by anyone.